test/lib/jdk/test/lib/threaddump/ThreadDump.java

```diff
@@ -0,0 +1,370 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.test.lib.threaddump;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.Set;
+import java.util.stream.Stream;
+import jdk.test.lib.json.JSONValue;
+
+/**
+ * Represents a thread dump that is obtained by parsing JSON text. A thread dump in JSON
+ * format is generated with the {@code com.sun.management.HotSpotDiagnosticMXBean} API or
+ * using {@code jcmd <pid> Thread.dump_to_file -format=json <file>}.
+ *
+ * <p> The following is an example thread dump that is parsed by this class. Many of the
+ * objects are collapsed to reduce the size.
+ *
+ * <pre>{@code
+ * {
+ *   "threadDump": {
+ *     "processId": "63406",
+ *     "time": "2022-05-20T07:37:16.308017Z",
+ *     "runtimeVersion": "19",
+ *     "threadContainers": [
+ *       {
+ *         "container": "<root>",
+ *         "parent": null,
+ *         "owner": null,
+ *         "threads": [
+ *          {
+ *            "tid": "1",
+ *            "name": "main",
+ *            "stack": [...]
+ *          },
+ *          {
+ *            "tid": "8",
+ *            "name": "Reference Handler",
+ *            "stack": [
+ *               "java.base\/java.lang.ref.Reference.waitForReferencePendingList(Native Method)",
+ *               "java.base\/java.lang.ref.Reference.processPendingReferences(Reference.java:245)",
+ *               "java.base\/java.lang.ref.Reference$ReferenceHandler.run(Reference.java:207)"
+ *            ]
+ *          },
+ *          {"name": "Finalizer"...},
+ *          {"name": "Signal Dispatcher"...},
+ *          {"name": "Common-Cleaner"...},
+ *          {"name": "Monitor Ctrl-Break"...},
+ *          {"name": "Notification Thread"...}
+ *         ],
+ *         "threadCount": "7"
+ *       },
+ *       {
+ *         "container": "ForkJoinPool.commonPool\/jdk.internal.vm.SharedThreadContainer@56aac163",
+ *         "parent": "<root>",
+ *         "owner": null,
+ *         "threads": [...],
+ *         "threadCount": "1"
+ *       }
```

Comment:
Why is there no longer a ',' here?

Reply:
I had to collapse several objects and reduce the number of items in the arrays in order to get it down to an example that fits into the class description. To yes, all items except the last should have a trailing comma.